src/Parsing/Parser.hs

```diff
@@ -21,8 +20,25 @@ module Parsing.Parser
 , phpParser
 , phpASTParser
 , haskellParser
-  -- Precise parsers
-, precisePythonParser
+  -- * Abstract parsers
+, SomeParser(..)
+, goParser'
+, haskellParser'
+, javascriptParser'
+, jsonParser'
+, jsxParser'
+, markdownParser'
+, phpParser'
+, pythonParserALaCarte'
+, pythonParserPrecise'
+, pythonParser'
+, rubyParser'
+, tsxParser'
+, typescriptParser'
+  -- * Canonical sets of parsers
+, aLaCarteParsers
+, preciseParsers
+, allParsers
```

Comment:
An idea: split this module up into `Parsing.ALaCarte`, `Parsing.Precise`, and `Parsing.Abstract`? That way you don’t need all these ticks.

Reply:
Well, we’d end up having to import more things in a bunch of places (`Parsing.Parser` couldn’t import those since they’d need `SomeParser` & `Parser`), and at the end of the day, the goal is for all the unticked versions to go away. I’d rather leave the ticks as an incentive to remove the à la carte parsers altogether 😊